### PR TITLE
[4.0] RavenDB-9643 Ensure an unique name of a temp export file for the offl…

### PR DESCRIPTION
**Pull request by arekpalinski**
…ine migration

### CODE REVIEW

File: src/Raven.Client/ServerWide/Operations/Migration/OfflineMigrationConfiguration.cs [hunk 1]
```diff
@@ -57,7 +57,7 @@ internal void Validate()
             if (OutputFilePath == null)
             {
                 var rempDir = Path.GetTempPath();
-                OutputFilePath = tmpFile = Path.Combine(rempDir, "export.ravendump");
+                OutputFilePath = tmpFile = Path.Combine(rempDir, $"export-{Guid.NewGuid()}.ravendump");
```
Comment (ayende) [inline src/Raven.Client/ServerWide/Operations/Migration/OfflineMigrationConfiguration.cs]: Use db name and date as well here, I think

Comment (arekpalinski) [reply src/Raven.Client/ServerWide/Operations/Migration/OfflineMigrationConfiguration.cs]: I've dropped the usage of Guid in favor of `$"export-{DatabaseRecord.DatabaseName}-{SystemTime.UtcNow:HH_mm_ss.ffff_MM-dd-yyyy}.ravendump"`

Comment (ml054) [reply src/Raven.Client/ServerWide/Operations/Migration/OfflineMigrationConfiguration.cs]: Isn't better to have yyyy-dd-MM at the beginning? (allows to sort by date)

Comment (arekpalinski) [reply src/Raven.Client/ServerWide/Operations/Migration/OfflineMigrationConfiguration.cs]: Yeah, I will fix it and make it more consistent with RavenDB-11126

**Pull request merged**